GPU/TPCSpaceChargeBase/AliTPCSpaceCharge3DCalc.cxx

```diff
@@ -2453,32 +2562,33 @@ void AliTPCSpaceCharge3DCalc::IntegrateDistCorrDriftLineDz(
           }
 
           //lookupLocalDist->GetValue(radius, phi, z, ddR, ddRPhi, ddZ);
-          localIntErOverEz = (intDrDzF->Eval(radius, phi, z1) - intDrDzF->Eval(radius, phi, z)) / (-1 * ezField);
-          localIntEPhiOverEz = (intDPhiDzF->Eval(radius, phi, z1) - intDPhiDzF->Eval(radius, phi, z)) / (-1 * ezField);
+          localIntErOverEz = (intDrDzF->Eval(radius, phi, z1) - intDrDzF->Eval(radius, phi, z)) / (ezField + ezF->Eval(radius, phi, z));
+          localIntEPhiOverEz = (intDPhiDzF->Eval(radius, phi, z1) - intDPhiDzF->Eval(radius, phi, z)) / (ezField + ezF->Eval(radius, phi, z));
           localIntDeltaEz = intDzDzF->Eval(radius, phi, z1) - intDzDzF->Eval(radius, phi, z);
 
           ddR = fC0 * localIntErOverEz + fC1 * localIntEPhiOverEz;
           ddRPhi = fC0 * localIntEPhiOverEz - fC1 * localIntErOverEz;
-          ddZ = localIntDeltaEz * AliTPCPoissonSolver::fgkdvdE * AliTPCPoissonSolver::fgkdvdE; // two times?
+          ddZ = -1 * localIntDeltaEz * AliTPCPoissonSolver::fgkdvdE;
 
-          // add local distortion
           drDist += ddR;
-          dRPhi += ddRPhi;
+          dPhi += (ddRPhi / radius);
           dzDist += ddZ;
+
+          // add local distortion
         }
         // set the global distortion after following the electron drift
         (*mDistDrDz)(i, j) = drDist;
-        (*mDistDPhiRDz)(i, j) = dRPhi;
+        (*mDistDPhiRDz)(i, j) = dPhi * radius0;
         (*mDistDz)(i, j) = dzDist;
         /////////////// use irregular grid look up table for correction
-        // set
+        // use oppsite directions of distortion
         (*mCorrIrregularDrDz)(i, j) = -drDist;
-        (*mCorrIrregularDPhiRDz)(i, j) = -dRPhi;
+        (*mCorrIrregularDPhiRDz)(i, j) = -dPhi * radius0;
```

Comment:
Same thing. As you use the distorted points as the reference points for the irregular correction lookup tables, the correction in rphi is -1 times distortion in phi (-dPhi) times the distorted radius (radius0 + drDist)